posthog/session_recordings/queries/sub_queries/base_query.py

```diff
@@ -6,6 +6,57 @@
 from posthog.models import Team
 from posthog.schema import RecordingsQuery, DateRange
 from posthog.session_recordings.queries.session_replay_events import ttl_days
+from posthog.utils import relative_date_parse
+
+
+class SessionRecordingsQueryDateRange(QueryDateRange):
+    """Custom QueryDateRange that only applies start/end of day logic when dates don't specify time components."""
+
+    def _has_time_component(self, date_str: str) -> bool:
+        """Check if a date string contains a time component."""
```

Comment:
```suggestion
```

stylistically i'm not a fan of comments that mostly duplicate the method name
(not blocking though... style comes and goes 😊)